### PR TITLE
xtensa/esp32: Add high memory support to work with PSRAM

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Add high memory support to work with PSRAM
## Impact
Users will be able to use more than 4MiB of PSRAM
## Testing
esp32-core

PS.: the check will fail because mixed functions defined in the ROM

### CODE REVIEW

File: arch/xtensa/src/esp32/esp32_psram.c [hunk 1]
```diff
@@ -1530,7 +1530,7 @@ psram_enable(int mode, int vaddrmode)   /* psram init */
         PSRAM_IS_64MBIT_TRIAL(s_psram_id)))
     {
 #if defined(CONFIG_ESP32_SPIRAM_2T_MODE)
-#  if CONFIG_ESP32_SPIRAM_BANKSWITCH_ENABLE
+#  if CONFIG_SPIRAM_BANKSWITCH_ENABLE
```
Comment (royfengsss) [inline arch/xtensa/src/esp32/esp32_psram.c:1533]: CONFIG_SPIRAM_BANKSWITCH_ENABLE or CONFIG_ESP32_SPIRAM_BANKSWITCH_ENABLE?
In arch/xtensa/src/esp32/Kconfig, this configure option changed to ESP32_SPIRAM_BANKSWITCH_ENABLE

Please kindly check.

Thanks

File: boards/xtensa/esp32/esp32-core/src/esp32_bringup.c [hunk 2]
```diff
@@ -143,7 +143,7 @@ int esp32_bringup(void)
 
 #endif
 
-#if defined(CONFIG_ESP32_SPIRAM)
+#if defined(CONFIG_ESP32_SPIRAM) && defined(CONFIG_SPIRAM_BANKSWITCH_ENABLE)
```
Comment (royfengsss) [inline boards/xtensa/esp32/esp32-core/src/esp32_bringup.c:146]: CONFIG_SPIRAM_BANKSWITCH_ENABLE or CONFIG_ESP32_SPIRAM_BANKSWITCH_ENABLE?
In _arch/xtensa/src/esp32/Kconfig_, this configure option changed to ESP32_SPIRAM_BANKSWITCH_ENABLE

Please kindly check.

Thanks

Comment (acassis) [reply boards/xtensa/esp32/esp32-core/src/esp32_bringup.c:146]: Thank you! I was going to replace CONFIG_SPIRAM_BANKSWITCH_ENABLE with CONFIG_ESP32_SPIRAM_BANKSWITCH_ENABLE but I regret on it and forgot to remove it from a place in the Kconfig.

**Review by Ouss4 — COMMENTED** [review]
Please squash the two last commits they address the same issue.

Regarding the prefix, I think adding ESP32 is the correct one not removing it.

**Comment by acassis** [comment]
CI is failing:

Run actions/upload-artifact@v2
With the provided path, there will be 121 file(s) uploaded
An error has been caught http-client index 0, retrying the upload
Error: read ECONNRESET
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:201:27) {
  errno: 'ECONNRESET',
  code: 'ECONNRESET',
  syscall: 'read'
}
Exponential backoff for retry #1. Waiting for 6066.758889064973 milliseconds before continuing the upload at offset 0
An error has been caught http-client index 1, retrying the upload
Error: Client has already been disposed.
    at HttpClient.request (/home/runner/work/_actions/actions/upload-artifact/v2/dist/index.js:5600:19)
    at HttpClient.sendStream (/home/runner/work/_actions/actions/upload-artifact/v2/dist/index.js:5561:21)
    at UploadHttpClient.<anonymous> (/home/runner/work/_actions/actions/upload-artifact/v2/dist/index.js:7008:37)
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/actions/upload-artifact/v2/dist/index.js:6742:71
    at new Promise (<anonymous>)
    at module.exports.608.__awaiter (/home/runner/work/_actions/actions/upload-artifact/v2/dist/index.js:6738:12)
    at uploadChunkRequest (/home/runner/work/_actions/actions/upload-artifact/v2/dist/index.js:7006:46)
    at UploadHttpClient.<anonymous> (/home/runner/work/_actions/actions/upload-artifact/v2/dist/index.js:7043:38)
    at Generator.next (<anonymous>)
Exponential backoff for retry #1. Waiting for 6496.523714895135 milliseconds before continuing the upload at offset 0
Total file count: 121 ---- Processed file #107 (88.4%)
Finished backoff for retry #1, continuing with upload
Finished backoff for retry #1, continuing with upload
Total size of all the files uploaded is 10763969 bytes
Error: read ECONNRESET

**Pull request merged**